org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/OpenQuoteExpectedCodeAction.java

```diff
@@ -43,10 +44,22 @@ public void doCodeAction(ICodeActionRequest request, List<CodeAction> codeAction
 		} catch (BadLocationException e) {
 			return;
 		}
-		DOMAttr attr = document.findAttrAt(offset);
-		if (attr == null || !attr.isAttribute()) {
-			return;
+		DOMNode attr = document.findAttrAt(offset);
+		if (attr != null && attr.isAttribute()) {
+			// ex : <foo bar=value > <-- error on value which must be quoted.
+			insertQuotationForAttr(attr, document, request, diagnostic, codeActions);
+		} else {
+			DTDDeclParameter parameter = document.findDTDDeclParameterAt(offset);
+			if (parameter != null) {
+				// ex : <!ATTLIST dadesadministratives idinstitut ID > <-- error on idinstitut
+				// which must be quoted.
+				insertQuotationForParameter(parameter, document, request, diagnostic, codeActions);
```

Comment:
in all cases of `Open quote is expected for attribute "attributename" associated with an  element type  "elementname".` errors, none of them are about missing a quote, but needing to insert `#REQUIRED` at the end of the declaration. 

So at least you should add that as another quickfix

Reply:
Indeed code actions is bad, I disable it.